components/shared/ui/appPage.tsx

```diff
@@ -5,31 +5,79 @@ import { perCollectionSEOTitle } from "../../../lib/constants/labels";
 import { ValidCollectionCodename } from "../../../lib/types/perCollection";
 import { useSmartLink } from "../../../lib/useSmartLink";
 import { createItemSmartLink } from "../../../lib/utils/smartLinkUtils";
-import { Block_Navigation } from "../../../models";
+import { Article, Block_Navigation, Product, SEOMetadata, WSL_WebSpotlightRoot } from "../../../models";
 import { SiteCodenameProvider } from "../siteCodenameContext";
 import { Footer } from "./footer";
 import { Menu } from "./menu";
 
 type Props = Readonly<{
   children: ReactNode;
   siteCodename: ValidCollectionCodename;
-  itemId?: string;
+  item?: WSL_WebSpotlightRoot | Article | Product;
   siteMenu?: Block_Navigation;
+  defaultMetadata?: SEOMetadata;
+  pageType: "Article" | "WebPage" | "Product";
 }>;
 
 export const AppPage: FC<Props> = props => {
   useSmartLink();
+
+  const getPageTitle = (): string => {
+    const siteTitle = perCollectionSEOTitle[props.siteCodename];
+    const titleExtension = props.item?.elements.seoMetadataTitle.value;
+
+    return titleExtension && titleExtension !== siteTitle ? `${siteTitle} | ${titleExtension}` : siteTitle;
+  };
+
+  const pageMetaDescription = props.item?.elements.seoMetadataDescription.value ?? props.defaultMetadata?.elements.seoMetadataDescription.value;
```

Comment:
I am not sure what is the value of a text element when it is empty, but this fallback (to `defaultMetadata`) won't work when there will be an empty string in the `item`. You can even `.trim()` it and use `||` to ignore empty (even white-spaced) element values.